### ISSUE TITLE
C++ error when clicking anywhere in the editor

### ISSUE THREAD

**Issue by anniryynanen** [godotengine/godot#93565]
### Tested versions

Reproduces in v4.3.beta2.official [b75f0485b]

I've used v4.3.beta1 a lot and I've never seen it there, so I'm guessing it's new in beta2.

### System information

Windows 10.0.22631 - GLES3 (Compatibility) - AMD Radeon RX 7700 XT (Advanced Micro Devices, Inc.; 31.0.24033.1003) - AMD Ryzen 7 5700X3D 8-Core Processor (16 Threads)

### Issue description

After using the editor for some time, clicking anywhere starts to print

```
 scene/gui/tree.cpp:1355 - Index p_column = 0 is out of bounds (cells.size() = 0).
```

Once it starts, all clicks in the editor print the error until I switch to another scene.

### Steps to reproduce

Not sure. I seem to get it eventually, I'll run the editor in debugger and maybe I can get more information when it happens the next time.

### Minimal reproduction project (MRP)

N/A

**Comment by AeioMuch** [comment]
Hello !

I saw this happening here and there after this got merged : https://github.com/godotengine/godot/pull/91265 so must be because on the change on how the custom_color is set/cleared now, but I'm not too sure how to reproduce reliably.

It guess it's happening when the custom_color is cleared when the mouse button is released but `tree_item_inspected` could get invalid if moved or deleted, I guess just doing `tree_item_inspected = nullptr;` when the the mouse button is released or something could be enough but hard to be sure without a way to reproduce for sure.

**Comment by pmoosi** [comment]
Hey! I just want to say that I was able to reproduce it quite reliably by reparenting a node in the sidepanel by hovering over the other node until its color changes. Usually after doing that once or twice the error appeared (and sometimes the editor crashed?).

I just applied the fix suggested by @AeioMuch (#93576) and it seems to fix it. At least I cannot reproduce it anymore.

**Comment by akien-mga** [comment]
I can reproduce the bug following the instructions from @pmoosi in the current `master` branch (ac9181c666fc9a28e56c6fbf37d665687c1d722a). I also got a crash:
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.3.beta.custom_build (ac9181c666fc9a28e56c6fbf37d665687c1d722a)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib64/libc.so.6(+0x40710) [0x7f25679f8710] (??:0)
[2] CowData<TreeItem::Cell>::size() const (/home/akien/Godot/godot/./core/templates/cowdata.h:183)
[3] Vector<TreeItem::Cell>::size() const (/home/akien/Godot/godot/./core/templates/vector.h:94)
[4] TreeItem::clear_custom_color(int) (/home/akien/Godot/godot/./scene/gui/tree.cpp:1355 (discriminator 1))
[5] SceneTreeDock::_inspect_hovered_node() (/home/akien/Godot/godot/./editor/scene_tree_dock.cpp:83)
[6] void call_with_variant_args_helper<SceneTreeDock>(SceneTreeDock*, void (SceneTreeDock::*)(), Variant const**, Callable::CallError&, IndexSequence<>) (/home/akien/Godot/godot/./core/variant/binder_common.h:309)
[7] void call_with_variant_args<SceneTreeDock>(SceneTreeDock*, void (SceneTreeDock::*)(), Variant const**, int, Callable::CallError&) (/home/akien/Godot/godot/./core/variant/binder_common.h:419)
[8] CallableCustomMethodPointer<SceneTreeDock>::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/akien/Godot/godot/./core/object/callable_method_pointer.h:104)
[9] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (/home/akien/Godot/godot/./core/variant/callable.cpp:57)
[10] Object::emit_signalp(StringName const&, Variant const**, int) (/home/akien/Godot/godot/./core/object/object.cpp:1188)
[11] Node::emit_signalp(StringName const&, Variant const**, int) (/home/akien/Godot/godot/./scene/main/node.cpp:3889)
[12] Error Object::emit_signal<>(StringName const&) (/home/akien/Godot/godot/./core/object/object.h:936)
[13] Timer::_notification(int) (/home/akien/Godot/godot/./scene/main/timer.cpp:62)
[14] Timer::_notificationv(int, bool) (/home/akien/Godot/godot/./scene/main/timer.h:37 (discriminator 14))
[15] Object::notification(int, bool) (/home/akien/Godot/godot/./core/object/object.cpp:873)
[16] SceneTree::_process_group(SceneTree::ProcessGroup*, bool) (/home/akien/Godot/godot/./scene/main/scene_tree.cpp:961)
[17] SceneTree::_process(bool) (/home/akien/Godot/godot/./scene/main/scene_tree.cpp:1034 (discriminator 2))
[18] SceneTree::process(double) (/home/akien/Godot/godot/./scene/main/scene_tree.cpp:528)
[19] Main::iteration() (/home/akien/Godot/godot/main/main.cpp:4099 (discriminator 3))
[20] OS_LinuxBSD::run() (/home/akien/Godot/godot/platform/linuxbsd/os_linuxbsd.cpp:962 (discriminator 1))
[21] /home/akien/Godot/godot/bin/godot.linuxbsd.editor.dev.x86_64(main+0x14b) [0x5b280b1] (/home/akien/Godot/godot/platform/linuxbsd/godot_linuxbsd.cpp:85)
[22] /lib64/libc.so.6(+0x2a088) [0x7f25679e2088] (??:0)
[23] /lib64/libc.so.6(__libc_start_main+0x8b) [0x7f25679e214b] (??:0)
[24] /home/akien/Godot/godot/bin/godot.linuxbsd.editor.dev.x86_64(_start+0x25) [0x5b27ea5] (??:?)
-- END OF BACKTRACE --
================================================================
```
